modules/external-api/src/main/java/org/opencastproject/external/endpoint/WorkflowsEndpoint.java

```diff
@@ -0,0 +1,629 @@
+          @RestResponse(description = "The request is invalid or inconsistent.", responseCode = HttpServletResponse.SC_BAD_REQUEST),
+          @RestResponse(description = "The user doesn't have the rights to make this request.", responseCode = HttpServletResponse.SC_FORBIDDEN),
+          @RestResponse(description = "The event, workflow definition or specified parent workflow could not be found.", responseCode = HttpServletResponse.SC_NOT_FOUND) })
+  public Response createWorkflowInstance(@HeaderParam("Accept") String acceptHeader,
+          @FormParam("event_identifier") String eventId,
+          @FormParam("workflow_definition_identifier") String workflowDefinitionIdentifier,
+          @FormParam("parent_identifier") Long parentIdentifier, @FormParam("configuration") String configuration,
+          @QueryParam("withoperations") boolean withOperations,
+          @QueryParam("withconfiguration") boolean withConfiguration) {
+    if (isBlank(eventId)) {
+      return RestUtil.R.badRequest("Required parameter 'event_identifier' is missing or invalid");
+    }
+
+    if (isBlank(workflowDefinitionIdentifier)) {
+      return RestUtil.R.badRequest("Required parameter 'workflow_definition_identifier' is missing or invalid");
+    }
+
+    try {
+      // Parent workflow
+      if (parentIdentifier != null) {
+        try {
+          workflowService.getWorkflowById(parentIdentifier);
+        } catch (NotFoundException e) {
+          return ApiResponses.notFound("Cannot find the parent workflow with id '%s'.", workflowDefinitionIdentifier);
+        }
+      }
+
+      // Media Package
+      Opt<Event> event = indexService.getEvent(eventId, externalIndex);
+      if (event.isNone()) {
+        return ApiResponses.notFound("Cannot find an event with id '%s'.", eventId);
+      }
+      MediaPackage mp = indexService.getEventMediapackage(event.get());
+
+      // Workflow definition
+      WorkflowDefinition wd;
+      try {
+        wd = workflowService.getWorkflowDefinitionById(workflowDefinitionIdentifier);
+      } catch (NotFoundException e) {
+        return ApiResponses.notFound("Cannot find a workflow definition with id '%s'.", workflowDefinitionIdentifier);
+      }
+
+      // Configuration
+      Map<String, String> properties = new HashMap<>();
+      if (isNoneBlank(configuration)) {
+        JSONParser parser = new JSONParser();
+        try {
+          properties.putAll((JSONObject) parser.parse(configuration));
+        } catch (ParseException e) {
+          return RestUtil.R.badRequest("Passed parameter 'configuration' is invalid JSON.");
+        }
+      }
+
+      // Start workflow
+      WorkflowInstance wi = workflowService.start(wd, mp, parentIdentifier, properties);
+      return ApiResponses.Json.created(acceptHeader, URI.create(getWorkflowUrl(wi.getId())),
+              workflowInstanceToJSON(wi, withOperations, withConfiguration));
+    } catch (UnauthorizedException e) {
```

Comment:
You might want to add...

```
    } catch (IllegalStateException e) {
      final ApiVersion requestedVersion = ApiMediaType.parse(acceptHeader).getVersion();
      return ApiResponses.Json.conflict(requestedVersion, obj(f("message", v(getMessage(e), BLANK))));
    }
```

Otherwise, clients receive a HTTP 500 Server Error in case the workflow could not be started because a workflow is already running for the event.

Example response for the code above in that situation:

`{"message":"IllegalStateException: Can't start workflow 'Fast Testing Workflow' for media package '8d3c8cf0-99e9-4821-9f4b-dec95598c7e2' because another workflow is currently active."}`